### PR TITLE
Backport v18 vijay new

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
The build had passed locally on Ubuntu on WSL, checking the Pipeline issue

**Comment by vijaydasmp** [comment]
(Pipeline is failing with message : There has been a runner system failure, please try again)

File: src/init.cpp [hunk 1]
```diff
@@ -127,6 +129,29 @@ static CDSNotificationInterface* pdsNotificationInterface = nullptr;
...
+ */
+#ifndef WIN32
+static const char* BITCOIN_PID_FILENAME = "dashd.pid";
+
+static fs::path GetPidFile()
+{
+    return AbsPathForConfigVal(fs::path(gArgs.GetArg("-pid", BITCOIN_PID_FILENAME)));
+}
+
+NODISCARD static bool CreatePidFile()
```
Comment (UdjinM6) [inline src/init.cpp:143]: 15278:
```suggestion
[[nodiscard]] static bool CreatePidFile()
```

File: src/wallet/rpcdump.cpp [hunk 2]
```diff
@@ -1352,6 +1352,7 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
             "      \"internal\": <true>                                    , (boolean, optional, default: false) Stating whether matching outputs should be treated as not incoming payments\n"
             "      \"watchonly\": <true>                                   , (boolean, optional, default: false) Stating whether matching outputs should be considered watched even when they're not spendable, only allowed if keys are empty\n"
             "      \"label\": <label>                                      , (string, optional, default: '') Label to assign to the address (aka account name, for now), only allowed with internal=false\n"
+            "      \"keypool\": <false>                                    , (boolean, optional, default: false) Stating whether imported public keys should be added to the keypool for when users request new addresses. Only allowed when wallet private keys are disabled\n"
```
Comment (UdjinM6) [inline src/wallet/rpcdump.cpp:1355]: 15426: we don't have this yet. 15426 requires at least 14075 (which requires others it seems), so it's too early - should drop 15426 for now

File: test/functional/wallet_txn_clone.py [hunk 3]
```diff
@@ -61,19 +63,14 @@ def run_test(self):
...
-        pos0 = 2 * (4 + 1 + 36 + 1 + 4 + 1)
-        hex400 = "00902f5009000000"
-        output_len = 16 + 2 + 2 * int("0x" + clone_raw[pos0 + 16:pos0 + 16 + 2], 0)
-        if (rawtx1["vout"][0]["value"] == 400 and clone_raw[pos0:pos0 + 16] != hex400 or rawtx1["vout"][0]["value"] != 400 and clone_raw[pos0:pos0 + 16] == hex400):
-            output0 = clone_raw[pos0:pos0 + output_len]
-            output1 = clone_raw[pos0 + output_len:pos0 + 2 * output_len]
-            clone_raw = clone_raw[:pos0] + output1 + output0 + clone_raw[pos0 + 2 * output_len:]
+        clone_tx = CTransaction()
+        clone_tx.deserialize(io.BytesIO(bytes.fromhex(clone_raw)))
+        if (rawtx1["vout"][0]["value"] == 40 and clone_tx.vout[0].nValue != 40*COIN or rawtx1["vout"][0]["value"] != 40 and clone_tx.vout[0].nValue == 40*COIN):
```
Comment (UdjinM6) [inline test/functional/wallet_txn_clone.py:68]: 15397: 400

File: src/util/system.cpp [hunk 4]
```diff
@@ -96,7 +96,7 @@ const std::string gCoinJoinName = "CoinJoin";
 int nWalletBackups = 10;
 
 const char * const BITCOIN_CONF_FILENAME = "dash.conf";
-const char * const BITCOIN_PID_FILENAME = "dashd.pid";
+
```
Comment (UdjinM6) [inline src/util/system.cpp:99]: 15278:
```suggestion
```

File: test/lint/lint-locale-dependence.sh [hunk 5]
```diff
@@ -15,6 +15,8 @@ KNOWN_VIOLATIONS=(
     "src/governance/governance-validators.cpp.*tolower"
     "src/httprpc.cpp.*trim"
     "src/init.cpp:.*atoi"
+    "src/init.cpp:.*fprintf"
+    "src/init.cpp:.*fprintf"
```
Comment (UdjinM6) [inline test/lint/lint-locale-dependence.sh:19]: 15278:
```suggestion
```

File: src/test/validation_chainstatemanager_tests.cpp [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    exp_tip = c1.m_chain.Tip();
+    BOOST_CHECK_EQUAL(validated_tip, exp_tip);
+
+    // Let scheduler events finish running to avoid accessing memory that is going to be unloaded
+    SyncWithValidationInterfaceQueue();
+
+    WITH_LOCK(::cs_main, manager.Unload());
+}
+
+BOOST_AUTO_TEST_SUITE_END()
```
Comment (UdjinM6) [inline src/test/validation_chainstatemanager_tests.cpp:107]: 18615: skipped at least 17737

File: src/test/txindex_tests.cpp [hunk 7]
```diff
@@ -72,10 +72,8 @@ BOOST_FIXTURE_TEST_CASE(txindex_initial_sync, TestChain100Setup)
     // shutdown sequence (c.f. Shutdown() in init.cpp)
     txindex.Stop();
 
-    threadGroup.interrupt_all();
-    threadGroup.join_all();
-
-    // Rest of shutdown sequence and destructors happen in ~TestingSetup()
+    // Let scheduler events finish running to avoid accessing any memory related to txindex after it is destructed
+    SyncWithValidationInterfaceQueue();
```
Comment (UdjinM6) [inline src/test/txindex_tests.cpp:76]: 18615: skipped 18234

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcdump.cpp (hunk 2 above)
Comment (vijaydasmp) [reply src/wallet/rpcdump.cpp]: dropped 15426

File: src/test/txindex_tests.cpp (hunk 7 above)
Comment (vijaydasmp) [reply src/test/txindex_tests.cpp]: dropped 18615

**Comment by vijaydasmp** [comment]
Hello @UdjinM6, Thanks, I have implemented your comments

**Review by UdjinM6 — APPROVED** [review]
LGTM 👍 

utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**